### PR TITLE
Fix stock-perp backtest resilience and BTC fallback

### PR DESCRIPTION
**Pull request by msei99**
## Summary
- allow expected TradFi stock market closure gaps for `xyz:*` coins when TradFi provider is configured
- avoid rejecting TradFi-backed stock-perp historical ranges due to pre-inception clipping
- add BTC/USD fallback fetch (`binanceusdm`) when primary exchange returns empty benchmark data
- guard backtest analysis ratios against zero-division edge cases

## Tests
- added targeted regression tests in `tests/test_hlcv_preparation.py` and `tests/test_backtest_analysis.py`
- ran:
  - `pytest tests/test_hlcv_preparation.py::TestHLCVManagerGapHandling::test_tradfi_stock_perp_large_weekend_gap_accepted`
  - `pytest tests/test_hlcv_preparation.py::TestPrepareHLCVSBtcFallback::test_prepare_hlcvs_uses_binance_fallback_for_btc`
  - `pytest tests/test_backtest_analysis.py::test_process_forager_fills_handles_zero_pnl_division`
  - `pytest tests/test_hlcv_preparation.py tests/test_backtest_analysis.py -v`

### CODE REVIEW

File: src/hlcv_preparation.py [hunk 1]
```diff
@@ -690,8 +707,35 @@ async def prepare_hlcvs(config: dict, exchange: str, *, force_refetch_gaps: bool
 
         om.update_date_range(int(timestamps[0]), int(timestamps[-1]))
         btc_df = await om.get_ohlcvs("BTC")
+        btc_source_exchange = exchange
```
[the remark this answers is not in the document]
Comment (msei99) [reply src/hlcv_preparation.py:750]: Good catch — fixed in b9c75cec. I now log the effective BTC benchmark source and also include it in `mss["__meta__"]["btc_source_exchange"]` in the single-exchange path.

File: src/hlcv_preparation.py [hunk 2]
```diff
@@ -1017,32 +1067,50 @@ async def prepare_hlcvs_combined(
...
-        if btc_df.empty:
-            logging.error(
+        btc_candidates = [
+            exchanges_to_consider[0] if len(exchanges_to_consider) == 1 else "binanceusdm"
+        ]
+        if "binanceusdm" not in btc_candidates:
+            btc_candidates.append("binanceusdm")
+
+        btc_df = pd.DataFrame()
+        btc_source_exchange = None
```
[the remark this answers is not in the document]
Comment (msei99) [reply src/hlcv_preparation.py:1120]: Good catch — fixed in b9c75cec. I now log the effective BTC benchmark source and also include it in `mss["__meta__"]["btc_source_exchange"]` in the combined path.

**Review by msei99 — COMMENTED** [review]
Addressed Copilot review feedback in b9c75cec by logging and exposing btc_source_exchange in __meta__ for both single and combined paths. Marked both threads as resolved; ready for re-review.

**Comment by msei99** [comment]
Update: clean summary of the just-pushed fix.

- Source-dir path now uses the same gap-tolerance policy as fallback (including the stock-perp 4-day floor).
- Source-dir data is standardized with CandlestickManager standardize_gaps(strict=False) before return, matching fallback behavior.
- Regression tests were updated/added for small-gap source-dir usage and stock-perp weekend-like gaps.
- Targeted hlcv_preparation source-dir tests pass.

Result: source-dir accepted data no longer fails later due to stricter no-gap assumptions, and both paths now behave consistently.
